frontend/src/queries/nodes/DataTable/DataTable.tsx

```diff
@@ -62,10 +63,16 @@ const groupTypes = [
 
 let uniqueNode = 0
 
-export function DataTable({ query, setQuery, context, cachedResults }: DataTableProps): JSX.Element {
-    const [key] = useState(() => `DataTable.${uniqueNode++}`)
+export function DataTable({ uniqueKey, query, setQuery, context, cachedResults }: DataTableProps): JSX.Element {
+    const uniqueNodeKey = useState(() => uniqueNode++)
+    const [nodeLogicKey] = useState(() => `DataNode.${uniqueKey || uniqueNodeKey}`)
+    const [tableLogicKey] = useState(() => `DataTable.${uniqueNodeKey}`)
```

Comment:
Why not also give the table this new key?

Reply:
Yep, I think you're right. We didn't need it for the use case I was testing but makes much more sense to add it so that others can reference it in future